content/blog/2022/2022-08-30-handling-timezones-in-spring.md

```diff
@@ -0,0 +1,516 @@
+    private Integer id;
+
+    @Column(name = "date_str")
+    private String dateStr;
+
+    @Column(name = "date_time")
+    private Date date;
+
+    @Column(name = "local_date")
+    private LocalDate localDate;
+
+    @Column(name = "local_time")
+    private LocalTime localTime;
+
+    @Column(name = "local_datetime_dt")
+    private LocalDateTime localDateTime;
+
+    @Column(name = "local_datetime_ts")
+    private LocalDateTime localDateTimeTs;
+
+    @Column(name = "offset_datetime")
+    private OffsetDateTime offsetDateTime;
+
+    @Column(name = "zoned_datetime")
+    private ZonedDateTime zonedDateTime;
+
+    @Column(name = "created_at", nullable = false, updatable = false)
+    @CreationTimestamp
+    private OffsetDateTime createdAt;
+
+````
+
+### Understanding the server timezone setup
+
+{{% image alt="settings" src="images/posts/handling-timezones-in-spring/TZ.JPG" %}}
+
+As seen, the database should store the timestamp in UTC (+00:00). We will run our spring boot application to the custom timezones **Europe/London** and **Europe/Berlin**.
+Since these timezones have offset +01:00 and +02:00 respectively, we can easily compare the timestamps stored and retrieved.
+
+To start the Spring application in Europe/London timezone we specify the timezone in the arguments as :
+````text
+    mvnw clean verify spring-boot:run -Dspring-boot.run.jvmArguments="-Duser.timezone=Europe/London"
+````
+
+Similarly, to start the application in Europe/Berlin timezone we would specify the arguments as :
+````text
+    mvnw clean verify spring-boot:run -Dspring-boot.run.jvmArguments="-Duser.timezone=Europe/Berlin"
+````
+
+We have configured two endpoints in the controller class:
+- `http://localhost:8083/app/v1/timezones/default` stores the current date/time in the jvm arguments specified timezone.
+- `http://localhost:8083/app/v1/timezones/dst` stores a custom date/time in the jvm arguments specified timezone.
+
+{{% info title="`Daylight Savings Time`" %}}
+As on 8th September 2022, both the timezones **Europe/London** and **Europe/Berlin** are on DST. Their corresponding timezones are **British Summer Time (BST) (UTC+1)** and **Central European Summer Time (CEST) (UTC+2)**.
+Post 30th October 2022, the DST will end and they will be back to **Greenwich Mean Time (GMT) (UTC)** and **Central European Time (CET) (UTC+1)** respectively.
+{{% /info %}}
+
+### Comparing results from both timezones
+
+For Europe/London at the current date/time :
+
+{{% image alt="settings" src="images/posts/handling-timezones-in-spring/spring_tz_1.JPG" %}}
+{{% image alt="settings" src="images/posts/handling-timezones-in-spring/db_tz_1.JPG" %}}
+
+On comparison, we can make note of the following points:
+- `VARCHAR` representation of date in the database is not preferred, since it stores the date in the format it was sent. This could result in inconsistent date formats and make it difficult to convert back in the application.
```

Comment:
In the items of this list, can you refer to the database columns (from the previous image) you are talking about? Otherwise the reader has to guess which one you are talking about.